images/osm2pgsql/run.py

```diff
@@ -39,9 +44,65 @@ def pg_is_ready():
             log.info('pg is ready for connections')
     return ready
 
+
+def first_import():
+    log.info('starting first osm import')
+    os.system("osm2pgsql \
+                --create \
+                --slim \
+                -G \
+                --hstore \
+                --tag-transform-script /src/openstreetmap-carto.lua \
+                -C 2500 \
+                --number-processes 4 \
+                -S /src/openstreetmap-carto.style \
+                /src/first-osm-import.osm"
+              )
+    log.info('first import is done')
+
+
+def parse_integer_to_directory_number(integer):
+    return "{0:0=3d}".format(integer)
+
+
+def is_file_or_directory(path):
+    return os.path.exists(path)
+
+
+def get_file_from_replication_server(path):
+    url = '/'.join([REPLICATION_URL, path])
+    log.info(f"fetching file from {url}")
+    result = requests.get(url)
+    return (result.ok, result.text)
+
+
+def download(path: str, dest_folder: str):
+    url = '/'.join([REPLICATION_URL, path])
+
+    if not os.path.exists(dest_folder):
+        os.makedirs(dest_folder)  # create folder if it does not exist
+
+    # be careful with file names
+    filename = url.split('/')[-1].replace(" ", "_")
+    file_path = os.path.join(dest_folder, filename)
+
+    r = requests.get(url, stream=True)
+    if r.ok:
+        log.info(f"saving file to {os.path.abspath(file_path)}")
+        with open(file_path, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=1024 * 8):
+                if chunk:
+                    f.write(chunk)
+                    f.flush()
+                    os.fsync(f.fileno())
+    else:
+        log.error(f'file download failed with error code {r.status_code}')
+    return r.status_code
```

Comment:
what about `urllib`? 

https://stackoverflow.com/questions/19602931/basic-http-file-downloading-and-saving-to-disk-in-python

Reply:
`urllib.URLopener()` and `urllib.urlretrieve` from the page you linked are both deprecated.
https://docs.python.org/3/library/urllib.request.html#urllib.request.URLopener

maybe instead of writing the file in chunks, we can write it in one go like this:
```python
url = 'https://www.facebook.com/favicon.ico'
r = requests.get(url, allow_redirects=True)
open('facebook.ico', 'wb').write(r.content)
```